src/core/js/app.js

```diff
@@ -75,36 +75,49 @@ require([
             }
             
             Adapt.setupMapping();
-            
-            if (isLanguageChange) {
-
-                Adapt.trigger('app:languageChanged', language);
-
-                _.defer(function() {
-                    var startController = new StartController();
-                    var hash = '#/';
-
-                    if (startController.isEnabled()) {
-                        hash = startController.getStartHash(true);
-                    }
-                    
-                    Backbone.history.navigate(hash, { trigger: true, replace: true });
-                });
-            }
 
             try {
-                Adapt.trigger('app:dataReady');
+                Adapt.trigger('app:dataLoaded');
             } catch(e) {
                 outputError(e);
             }
 
-            Adapt.navigation = new NavigationView();// This should be triggered after 'app:dataReady' as plugins might want to manipulate the navigation
-            
-            Adapt.initialize();
-            
-            Adapt.off('adaptCollection:dataLoaded courseModel:dataLoaded');
+            if (!Adapt.isWaitingForPlugins()) triggerDataReady(isLanguageChange);
+            else Adapt.once('plugins:ready', function() {
+                triggerDataReady(isLanguageChange);
+            });
         }
     };
+
+    function triggerDataReady(isLanguageChange) {
+        if (isLanguageChange) {
+
+            Adapt.trigger('app:languageChanged', language);
```

Comment:
We need to look at the variables `isLanguageChange` and `language`.

`language` will be undefined here. Maybe instead of passing through the boolean `isLanguageChange`, we should pass through the actual `language` on line 196. This would be additionally be passed through to `triggerDataReady` on lines 85 & 87.

We may then drop `isLanguageChange` altogether and use `language` as a truthy value in the logic:
```js
function triggerDataReady(language) {
    if (language) {
        Adapt.trigger('app:languageChanged', language);
        ...
```